### ISSUE TITLE
Use data extracts hosted in raw-data-api

### ISSUE THREAD

**Issue by spwoodcock** [hotosm/fmtm#934]
## Issue

- Currently we keep track of data extracts within the FMTM database.
- We request a data extract from raw-data-api, then input the resulting geojson data into our db.
- When we make a request for a subset of the extract (i.e. a task area), it comes from the db.
- If we need to update the data extract, we request/download again & then update the db.

## Solution

All of the above can be handled by raw-data-api:

- Request a data extract to be generated in flatgeobuf with bind_zip=False (i.e. not zipped).
- Normally the data only lasts for 90 days, but if we use a filename of `hotosm-fmtm-project-{project_id}`, the data will be persisted.
- If we request another extract for the same file, the file will be updated in place.

Essentially we remove the burden of handling data extracts ourselves, and outsource it to raw-data-api.

Tasking manager is going to do something similar.

The resulting flatgeobuf has a spatial index inbuilt and can be accessed via RANGE request, so we can subset the data for a specific task area, the same way we do now.

To do this, we pass these params via POST to the `/snapshot` endpoint:

```json
{
    "outputType": "fgb",
    "fileName": "hotosm-fmtm-project-{project_id}",
    "bindZip": false,
    "geometry": {...},
    "filters": {...}
}
```

#### Environment

- We should also probably consider what environment we are doing this on: main, staging, development:

**main**: the filename can be `hotosm-fmtm-project-{project_id}` (will exist forever)
**staging**: the filename can be `hotosm-fmtm-staging-project-{project_id}` (will exist for undefined period, perhaps 180 days?)
**development**: the filename can be `fmtm-project-{project_id}` (will expire in 90 days with no `hotosm` prefix)

The final S3 path will be something like:

```
S3_URL/fmtm/{project_id}/hotosm-fmtm-project-{project_id}.fgb
```

#### Redundant Tables

- We can completely remove the database tables and models for managing data extracts once this refactor is complete.

- Currently we have DB tables DbOsmLines (ways) and DbBuildings.

- Do we need a flatgeobuf for each, or can we just filter in the frontend by a key? In the final parsed geojson we will have the property `building=true|false`.

**Comment by robsavoye** [comment]
Don't use raw-data-api directly, use osm-fieldwork instead. Make_data_extract.py now uses osm-rawdata to get the data from the raw-data-api, but also support the data filtering which is required by ODK Collect. I also see in the Done table that 877 and 879 claim this is already done ? Any changes to data extracts need to be tested with ODK Collect, cause they break very easily.

**Comment by spwoodcock** [comment]
Updated the title for clarity.

OSM-rawdata needs to be updated to support this option.
